src/net.h

```diff
@@ -764,9 +772,8 @@ class CNode
     }
     // This boolean is unusued in actual processing, only present for backward compatibility at RPC/QT level
     bool m_legacyWhitelisted{false};
-    bool fFeeler{false}; // If true this node is being used as a short lived feeler.
-    bool fOneShot{false};
-    bool m_manual_connection{false};
+    ConnectionType m_conn_type{ConnectionType::OUTBOUND};
+    bool fAddrFetch{false};
```

Comment:
I explored the option but there were enough call sites that I thought it would be unnecessarily verbose for no additional value. Since `fAddrFetch` is set via this comparison [here](https://github.com/bitcoin/bitcoin/pull/19316/files#diff-9a82240fe7dfe86564178691cc57f2f1R2755), its mostly a simpler way to reference & doesn't introduce room for divergence I believe. Although one thing I'll do to strengthen this guarantee is remove the initialization to `false` here & make it a `const`.

Reply:
I see, maybe also consider a method like
```cpp
bool IsAddrFetch() { return m_conn_type == ConnectionType::ADDR_FETCH; }
```

Reply:
is there a reason you prefer the function over member var? I haven't incorporated yet but am open to it. just trying to understand the reasoning better

Reply:
* A bunch of methods would define the interface better. Otherwise the interface is "I expose `m_conn_type` to you and you do whatever you want with it (including changing it). OTOH if methods are used `m_conn_type` can and should be made private.
* It is shorter at the caller site (and I believe more readable).
* Can have methods like `IsOutbound()` which do not translate to a single enum value and would need more complex expressions at the caller site.
* Then it would be possible to change the enum `m_conn_type` to a set of boolean flags, bitmask or whatever other ideas there might be without changing the external interface.

Reply:
ah, I think my question was unclear. let me try again- 

in the latest push, `m_conn_type` is indeed a [private member var](https://github.com/bitcoin/bitcoin/pull/19316/files#diff-9a82240fe7dfe86564178691cc57f2f1R2746) & I've introduced [several accessor functions](https://github.com/bitcoin/bitcoin/pull/19316/files#diff-1a8b9d1ad0a6fda5e751286c73102fc2R794) for callers.

My question here is asking about leaving `m_addr_fetch` as a member var on `CNode` VS another accessor function `IsAddrFetchConn()`. I'm wondering what the advantages are of adding the function vs leaving it in the current / past state of being accessed through this member var, `m_addr_fetch`. 

(same for inbound connection type)

Reply:
I suggested to use a method `IsAddrFetch()` because you said `m_conn_type == ConnectionType::ADDR_FETCH` would be unnecessarily verbose at the callers' sites (it is, I agree).

Now that accessor methods have been added I think there is no need to keep `fAddrFetch` and it can be replaced by `IsAddrFetch()`?

Reply:
removed `fAddrFetch` and introduced `IsAddrFetch()` in latest push 🎈